trackintel/analysis/tracking_quality.py

```diff
@@ -92,7 +93,9 @@ def temporal_tracking_quality(source, granularity="all", max_iter=60):
         # get the tracked day relative to the first day
         start_date = df["started_at"].min().date()
         df["day"] = df["started_at"].apply(lambda x: (x.date() - start_date).days)
-        dict_day_to_date = dict(zip(df["day"], df["started_at"].apply(lambda x: x.date())))
+        dict_start_day_to_date = dict(zip(df["day"], df["started_at"].apply(lambda x: x.date())))
+        dict_finish_day_to_date = dict(zip(df["day"], df["finished_at"].apply(lambda x: x.date())))
```

Comment:
why do we need two dict? if `granularity == "day"`, we first split the original df over multiple days, such that for each record `started_at` day == `finished_at` day. These two dict are essentially the same?

Reply:
no, they were not the same in my test case, one day was missing, so I used other dict to ensure all days are present, the previous [comment ](https://github.com/mie-lab/trackintel/pull/328#discussion_r719583534) addresses this. 
Summarily, we should agree on the test case.

Reply:
The problem with the test case is that the last finished_at is at midnight the next day, so if we calculate the quality according to both the started_at and finished_at time, we will always get the last day with 0 quality. 

So I think we should avoid using finished_at. Since we split the records before, the started_at time will always cover all the days (except for the last day with quality 0 in the extreme case )

Reply:
yes, this is what was led me to have that extra one day! thanks for looking into it!